pkg/logs/internal/util/containersorpods/containers_or_pods.go

```diff
@@ -121,6 +121,11 @@ func NewDecidedChooser(decision LogWhat) Chooser {
 
 // Wait implements Chooser#Wait.
 func (ch *chooser) Wait(ctx context.Context) LogWhat {
+	// Assume if ctx has an error the context is cancelled or something gone wrong so can't tell which should be used
+	if ctx.Err() != nil {
+		return LogUnknown
+	}
```

Comment:
I ended up adding this as was still getting a race condition being detected on the call to `ch.start()` a couple of lines down. This stopped it happening yet I've now removed it (locally) and can't get it to fail with a race detection.

Reply:
This particular hunk makes no sense -- if ctx.Err() is non-nil, then ctx.Done() is closed, so the select just below this one will return LogUnknown (or possibly the value in ch.choice, if a decision has also been made).

Reply:
I added the check to stop `ch.start()` (the next line bellow) getting called if the context was cancelled. The main reason is that `ch.start` then goes off and starts some more goroutines that access the main Datadog global config. I believe that `config.mock` is changing that global config while these goroutines are accessing it after the context has possibly been cancelled. Thinking about it, maybe what I should do here is actually check context before running `ch.start()`, that way it would still return the value in `ch.choice` if one exists or `LogUnknown`. Going to give that ago in another branch to check if still breaks or not (removing the check causes the tests to fail pretty regularly).

Reply:
Ah, I see that the access to config.Datadog is problematic.  That goroutine will immediately shutdown, but not before accessing config.Datadog.